pkg/proxy/proxy.go

```diff
@@ -349,7 +349,19 @@ func BootProxy(logger *zap.Logger, opt Option, appCmd, appContainer string, pid
 	if opt.Port == 0 {
 		opt.Port = 16789
 	}
+	maxAttempts := 1000
+	attemptsDone:=0
 
+
+    if !isPortAvailable(opt.Port) {
+		for i := 1024; i <= 65535 && attemptsDone < maxAttempts; i++ {
+			if isPortAvailable(uint32(i)) {
+				opt.Port = uint32(i)
+				break
```

Comment:
If all ports are busy you should throw an error right ..? missed this in last review my bad.